pkg/imgpkg/registry/keychain_test.go

```diff
@@ -664,6 +667,28 @@ func TestOrderingOfAuthOpts(t *testing.T) {
 	})
 }
 
+func registerDefaultDockerProvider() string {
+	tempConfigJSONProviderDir, err := ioutil.TempDir(os.TempDir(), "test-default-keychain-provider")
+	if err != nil {
+		panic(fmt.Errorf("unable to run test: %s", err))
+	}
+
+	err = ioutil.WriteFile(filepath.Join(tempConfigJSONProviderDir, "config.json"), []byte(`{
```

Comment:
im assuming we are adding so that if somehow its getting picked up tests would fail. if so, should we make this file invalid, so that it makes things blow up.

Reply:
It is a good idea but....

the `defaultDockerConfigProvider` won't blow up. it will just try the next path in its list of paths to try reading a docker config.json from.

On most dev machines and even our e2e tests on GH we will have a `~/.docker/config.json` and it will try to read and load that file instead.

With the above change made in this PR:
- every `CLI flag` test fails
- 2 of the ordering tests fail (one being that cli > config.json)